tests/test_internals/test_globbing_utils.py

```diff
@@ -0,0 +1,74 @@
+from hdmf.testing import TestCase
+
+from nwb_conversion_tools.utils import get_fstring_variable_names, get_fstring_values_from_filename
+
+
+class TestGlobbingAssertions(TestCase):
+    def test_get_fstring_variable_names_assertion(self):
+        with self.assertRaisesWith(
+            exc_type=AssertionError,
+            exc_msg=(
+                "Empty variable name detected in f-string! Please ensure there is text between all "
+                "enclosing '{' and '}'."
+            ),
+        ):
+            get_fstring_variable_names(fstring="a/{x}b{y}/c{z}d{}")
+
+        def test_get_fstring_separators_assertion(self):
+            with self.assertWarnsWith(
+                warn_type=UserWarning,
+                exc_msg=(
+                    "There is an empty separator between two variables in the f-string! "
+                    "The f-string will not be invertible."
+                ),
+            ):
+                get_fstring_variable_names(fstring="a/{x}{y}/c{z}")
+
+    def test_get_fstring_values_from_filename_non_invertible_assertion(self):
+        with self.assertRaisesWith(
+            exc_type=AssertionError,
+            exc_msg=("Adjacent variable values contain the separator character! The f-string is not invertible."),
```

Comment:
Not every important but I find the concept of invertible f-string hard to be not very transparent. That is, I can't really understand that assertion message. That said, I have no better option as I don't really know the context in which this will be used.

Reply:
I'll probably add extensive documentation of this kind of thing as an extra section in the YML tutorial. But the basic fix is for the user to rename their files to use more distinctive separations between variables in the f-string.